### DISCUSSION TITLE
How to properly clone a Git-LFS repo using libgit2?

### DISCUSSION THREAD

**Discussion by hwiedPro** [libgit2/libgit2#6789] [Q&A]
### Discussed in https://github.com/libgit2/libgit2/discussions/6551

<div type='discussions-op-text'>

<sup>Originally posted by **hwiedPro** April 18, 2023</sup>
Hi, I haven't yet found any instructions on how to properly checkout git repos that use the Large File Storage (LFS) using libgit2.
Only some issues that made me think it should basically be possible, but none stated on how to do it...

Please also see my SO question:
https://stackoverflow.com/questions/75733554/how-to-properly-clone-a-git-lfs-repo-with-libgit2

Thanks in advance</div>

**Comment by lrm29** [comment]
Have a look at https://github.com/libgit2/libgit2/blob/main/tests/libgit2/filter/wildcard.c

This is the approach I take, registering a wildcard filter and then taking the name of the filter and looking up smudge/clean in the config. You can get the "mode" (smudge/clean) from git_filter_source. Then you can execute the filter. There's nothing LFS specific about this approach.

I haven't worked out how to make use of the "process" filter yet - turns out this is very necessary for LFS to not be slow, particularly on Windows.

**Comment by hwiedPro** [comment]
@lrm29 Thanks for your answer again, do you have by any chance a minimal example?
Unfortunately for me the API documentation is not that self explanatory...

Thanks in advance! :)
